firmware/controllers/system/PeriodicController.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "ThreadController.h"
+#include "efitime.h"
+
+template <int TStackSize>
+class PeriodicController : public ThreadController<TStackSize>
+{
+private:
+    const systime_t m_period;
+    bool m_running = true;
+
+protected:
+    const float m_periodSeconds;
+
+    /**
+     * @brief Called before running the periodic task.  Optionally override this method to set up.
+     */
+    virtual void OnStarted() {};
+
+    /**
+     * @brief Called periodically.  Override this method to do work for your controller.
+     */
+    virtual void PeriodicTask(efitime_t nowNt) = 0;
+
+    /**
+     * Called when the controller is stopped.  Optionally override this method to clean up.
+     */
+    virtual void OnStopped() {};
```

Comment:
is there one single usage for this in rusEfi at the moment?
I know there are two ways of doing OOP - you can either develop nice API regardless of usages and focus on API niceness, or you can make sure to never have a line of dead code?

Reply:
I'm not sure--is there anywhere that something has to get reset or reinitialized at run time?

Reply:
If there's not I'm happy to remove it until such a need arises (or doesn't).

Reply:
@mck1117 I have a strong preference against implementing new unused stuff for the future. Unless you have a usage, please remove

Reply:
yep, removed